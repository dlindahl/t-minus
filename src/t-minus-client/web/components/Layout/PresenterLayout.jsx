import TimerProgress from '../Progress/TimerProgress'
import { PropTypes } from 'react'

const baseStyles = {
  app: {
    display: 'flex',
    flex: 1,
    flexDirection: 'column',
    height: '100vh'
  },
  header: {
    padding: 15
  },
  main: {
    display: 'flex',
    flex: 1,
    flexDirection: 'column'
  },
  root: {
    display: 'flex'
  }
}

const Layout = (props) => (
  <div style={baseStyles.root}>
    <div style={baseStyles.app}>
      <main style={baseStyles.main}>
        {props.children}
      </main>
      <TimerProgress/>
    </div>
  </div>
)

Layout.propTypes = {
  children: PropTypes.node
}

export default Layout
