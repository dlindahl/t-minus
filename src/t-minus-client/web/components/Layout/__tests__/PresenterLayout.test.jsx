import configureStore from 'redux-mock-store'
import emptyObj from 'empty/object'
import PresenterLayout from '../PresenterLayout'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'

const STATES = {
  default: {
    clock: emptyObj,
    display: emptyObj
  }
}
const store = configureStore()

describe('PresenterLayout', () => {
  it('renders the component', () => {
    const component = renderer.create(
      <Provider store={store(STATES.default)}>
        <PresenterLayout/>
      </Provider>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
