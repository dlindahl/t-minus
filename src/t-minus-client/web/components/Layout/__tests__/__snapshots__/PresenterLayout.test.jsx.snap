exports[`PresenterLayout renders the component 1`] = `
<div
  style={
    Object {
      "display": "flex",
    }
  }>
  <div
    style={
      Object {
        "display": "flex",
        "flex": 1,
        "flexDirection": "column",
        "height": "100vh",
      }
    }>
    <main
      style={
        Object {
          "display": "flex",
          "flex": 1,
          "flexDirection": "column",
        }
      } />
    <div
      style={
        Object {
          "background": undefined,
          "height": 50,
        }
      } />
  </div>
</div>
`;
